include/ftxui/component/loop.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef FTXUI_COMPONENT_LOOP_HPP
+#define FTXUI_COMPONENT_LOOP_HPP
+
+#include <memory>  // for shared_ptr
+
+#include "ftxui/component/component_base.hpp"  // for ComponentBase
+
+namespace ftxui {
+class ComponentBase;
+
+using Component = std::shared_ptr<ComponentBase>;
+class ScreenInteractive;
+
+class Loop {
+ public:
+  Loop(ScreenInteractive* screen, Component component);
```

Comment:
@niXman,
What do you think of this new API?

Reply:
perfect!

will try it today. thanks!

Reply:
@ArthurSonzogni 

this is strange, but when I exec the `custom_loop` example it only updates counters when I move mouse. is it supposed to be like that?

Reply:
The external loop is updated 100 times per seconds. However the value is drawn to the screen only when FTXUI draw a new frame. The previous frame is invalidated when a new ftxui event is processed (e.g. a mouse move or a keyboard action)

Does this sounds wrong to you?

Reply:
ah, got it!

> Does this sounds wrong to you?

no-no, please leave it as is.

thanks!

will try to use loop in that style:
```
auto screen = ScreenInteractive::FitComponent();

  // Get called periodically:
  screen.PostPeriodicTask([&] {
    // [...]
    return true; // Continue
  }, 200ms);

  auto component = ...;
  screen.Loop(component);
```

will report!

Reply:
ah, it's not ready yet :)